### PR TITLE
Add IsingXY

### PR DESCRIPTION
**Pull request by multiphaseCFD**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ x] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

Macbook air M1 doesn't support large memory allocation in the tests

**Related GitHub Issues:**

### CODE REVIEW

**Comment by multiphaseCFD** [comment]
> Yes, there is an issue due to my poorly written workflow 😞 . You can just push an empty commit by
> 
> ```
> git commit --allow-empty -m 'Trigger CI'
> git push
> ```
> 
> to rerun all workflows.

Done!

**Comment by chaeyeunpark** [comment]
It seems like all tests pass besides `Format (C++)`. After installing `clang-format-12`, you may just run `make format` in the root of source code directory.

**Comment by multiphaseCFD** [comment]
> clang-format-12

Thanks! I failed to install clang-format on M1 and I tried to use black to make better format. Will try to install clang-format again and get back to you.

**Comment by AmintorDusko** [comment]
Hi @multiphaseCFD. I think you need to add `.DS_store` to the `.gitignore` file. 
You may need to remove the files already committed to the PR.

File: pennylane_lightning/src/tests/Test_GateImplementations_Param.cpp [hunk 1]
```diff
@@ -411,6 +411,153 @@ void testApplyIsingXX() {
...
+            ComplexPrecisionT{0.255280800811, 0.161572331669},
+            ComplexPrecisionT{0.119218164572, 0.115460377284},
+            ComplexPrecisionT{-0.000315789761, 0.153835664378},
+            ComplexPrecisionT{0.206786872079, 0.157633689097},
+            ComplexPrecisionT{0.093027614553, 0.271012980118},
+            ComplexPrecisionT{0.041495874524, 0.065934829414},
+            ComplexPrecisionT{0.089653238654, 0.221581339836},
+            ComplexPrecisionT{0.217892318964, 0.291261285543},
+            ComplexPrecisionT{0.292993247509, 0.186570793390},
+        };
```
Comment (chaeyeunpark) [inline pennylane_lightning/src/tests/Test_GateImplementations_Param.cpp:551]: Is this the result from `default.qubit`?

**Review by chaeyeunpark — COMMENTED** [review]
I have one specific question here. I also would like to ask you benchmark two implementations. You can run `make gbenchmark` and follow the instruction in https://github.com/PennyLaneAI/pennylane-lightning/blob/master/pennylane_lightning/src/benchmarks/README.md#benchmarksbench_kernels. Note that you can also filter the benchmarks to run. For more information, you may run `./BuildGBench/benchmarks/bench_kernels --help`

File: pennylane_lightning/src/tests/Test_GateImplementations_Param.cpp (hunk 1 above)
Comment (multiphaseCFD) [reply pennylane_lightning/src/tests/Test_GateImplementations_Param.cpp:551]: > Is this the result from `default.qubit`?

Yes

**Review by AmintorDusko — COMMENTED** [review]
Nice work, so far!

**Comment by multiphaseCFD** [comment]
> I have one specific question here. I also would like to ask you benchmark two implementations. You can run `make gbenchmark` and follow the instruction in https://github.com/PennyLaneAI/pennylane-lightning/blob/master/pennylane_lightning/src/benchmarks/README.md#benchmarksbench_kernels. Note that you can also filter the benchmarks to run. For more information, you may run `./BuildGBench/benchmarks/bench_kernels --help`

Screen Shot 2022-06-14 at 2 37 03 PM

**Review by mlxd — APPROVED** [review]
Nice work @multiphaseCFD !
I'll approve. Once the others approve too we can merge this in.

**Comment by multiphaseCFD** [comment]
> Thanks @multiphaseCFD for a nice job! Still, I think you may need to implement a generator for a backward path. When a gate is a [one-paramter Lie group](https://en.wikipedia.org/wiki/One-parameter_group), given by e−iGθ, we call G as the generator. Could you try this too? Please let me know when there is any question.

Sure! I will try this after I finalized current PR. As @mlxd  told me, what I have to do for this PR is just implementing IsingXY gate. Is it fine with you @chaeyeunpark ?

**Comment by chaeyeunpark** [comment]
Sure! Looking forward to seeing the generator for IsingXY in a subsequent PR. I am also gonna approve this.

**Review by chaeyeunpark — APPROVED** [review]

**Review by AmintorDusko — APPROVED** [review]
Nice work! Thanks for that!

**Comment by multiphaseCFD** [comment]
> Sure! Looking forward to seeing the generator for IsingXY in a subsequent PR. I am also gonna approve this.

Sure! I've put it into my tasks.

**Comment by multiphaseCFD** [comment]
> Nice work! Thanks for that!

Thanks!

**Pull request merged**